pusher/src/Services/AdminWokaService.ts

```diff
@@ -0,0 +1,64 @@
+import axios from "axios";
+import { ADMIN_API_TOKEN, ADMIN_API_URL } from "../Enum/EnvironmentVariable";
+import { WokaDetailsResult, WokaList } from "../Enum/PlayerTextures";
+import { WokaServiceInterface } from "./WokaServiceInterface";
+
+class AdminWokaService implements WokaServiceInterface {
+    /**
+     * Returns the list of all available Wokas for the current user.
+     */
+    getWokaList(token: string): Promise<WokaList | undefined> {
+        return axios
+            .get(`${ADMIN_API_URL}/api/woka-list/${token}`, {
+                headers: { Authorization: `${ADMIN_API_TOKEN}` },
+            })
+            .then((res) => {
+                return res.data;
+            })
+            .catch((err) => {
+                console.error(`Cannot get woka list from admin API with token: ${token}`, err);
+                return undefined;
+            });
+    }
+
+    /**
+     * Returns the URL of all the images for the given texture ids.
+     *
+     * Key: texture id
+     * Value: URL
+     *
+     * If one of the textures cannot be found, undefined is returned
+     */
+    fetchWokaDetails(textureIds: string[]): Promise<WokaDetailsResult | undefined> {
+        return axios
+            .post(
+                `${ADMIN_API_URL}/api/woka-details`,
+                {
+                    textureIds,
+                },
+                {
+                    headers: { Authorization: `${ADMIN_API_TOKEN}` },
+                }
+            )
+            .then((res) => {
+                const result: WokaDetailsResult = res.data;
```

Comment:
res.data is "any" so we cast this to WokaDetailsResult without really being sure here.

Could you write a type guard for `WokaDetailsResult` (using generic-type-guards) ?